src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -0,0 +1,246 @@
+    root = turretMech.getRoot("Root", 2, 2);
+    turretAngleSim =
+        root.append(new MechanismLigament2d("Turret", 2, 0, 5, new Color8Bit(Color.kRed)));
+    turretTab.add("Turret", turretMech);
+
+    turretMotor.restoreFactoryDefaults();
+    turretMotor.setSmartCurrentLimit(40);
+
+    this.robotStateManager = robotStateManager;
+
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -2);
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 2);
+
+    turretMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, true);
+    turretMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kForward, true);
+
+    // initialze PID controller and encoder objects
+    turretPIDController =
+        new PIDController(
+            Constants.TurretConstants.KP,
+            Constants.TurretConstants.KI,
+            Constants.TurretConstants.KD);
+    m_encoder = new CANcoder(Constants.TurretConstants.CANcoder_ID);
+
+    simEncoder =
+        new SimDeviceSim("CANEncoder:CANCoder (v6)", Constants.TurretConstants.CANcoder_ID);
+    simTurretPos = simEncoder.getDouble("rawPositionInput");
+
+    zeroTurretEncoder();
+    turretPIDController.setIZone(Constants.TurretConstants.KIZ);
+  }
+
+  public void stopTurret() {
+    turretMotor.stopMotor();
+  }
+
+  public Command stowTurret() {
+    return new InstantCommand(() -> setTurretPos(Math.toRadians(0))).withName("StowTurretCommand");
+  }
+
+  public void setTurretPos(double setpoint) {
+    turretGoalPositionEntry.log(setpoint);
+    turretMotor.set(
+        -turretPIDController.calculate(
+            turretPosition,
+            MathUtil.clamp(
+                setpoint,
+                Math.toRadians(-Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES),
+                Math.toRadians(Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES))));
+  }
+
+  public void holdPosition() {
+    setTurretPos(turretPosition);
+  }
+
+  public void zeroTurretEncoder() {
+    m_encoder.setPosition(0.0);
+  }
+
+  public void updateTurretPosition() {
+    turretPosition =
+        Math.toRadians(
+            ((m_encoder.getPosition().getValueAsDouble()) * 360)
+                * Constants.TurretConstants.CONVERSION_FACTOR);
+    SmartDashboard.putNumber("Turret Position", turretPosition);
+    SmartDashboard.putBoolean("Out of Bounds", Math.abs(turretPosition) > 3.14);
+    SmartDashboard.putBoolean(
+        "Soft limit enabled forward", turretMotor.isSoftLimitEnabled(SoftLimitDirection.kForward));
+    SmartDashboard.putBoolean(
+        "Soft limit enabled reverse", turretMotor.isSoftLimitEnabled(SoftLimitDirection.kReverse));
+  }
+
+  public double getTurretPos() {
+    return turretPosition; // returns the absolute encoder position in radians
+  }
+
+  public double getTurretVel() {
+    return turretVelocity;
+  }
+
+  public Command testTurretCommand() {
+    return runEnd(() -> setTurretPos(Math.toRadians(-60)), this::stopTurret);
+  }
```

Comment:
This would be useful to set to a specific angle as we need. Maybe modify to make it take a angle parameter (i.e. `testTurretCommand(double degrees)`)